File: benchmark.py
# import the necessary packages
import numpy as np
import cv2

def threshold(color, color2, t):
	return (abs(color[0]-color2[0]) < t) and (abs(color[1]-color2[1]) < t) and (abs(color[2]-color2[2]) < t)
color = [255,255,0] #cyan

image = cv2.imread("benchmark.jpg") #read in
x = 0
y = 0
count = 0
for i in range(0,len(image)): #rows
	for j in range(0,len(image[i])): #cols
		if threshold(color,image[i,j]):
			x += j
			y += i
			count += 1
if count == 0:
	print ("(0,0)")
else:
	print ("(%d, %d)" % (x/count, y/count))
			

File: benchmark.cpp
#include <opencv2/core/core.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <stdio.h>
int threshold (uchar targetB, uchar targetG, uchar targetR, uchar b, uchar g, uchar r, int t ){
	if (MAX(targetB,b)-MIN(targetB,b) > t)
		return false;
	if (MAX(targetG,g)-MIN(targetG,g) > t)
		return false;
	if (MAX(targetR,r)-MIN(targetR,r) > t)
		return false;
	return true;
};
int main( int argc, char** argv )
{
	cv::Mat m;
	m = cv::imread("benchmark.jpg", CV_LOAD_IMAGE_COLOR);
	uchar r = 0;
	uchar g = 255;
	uchar b = 255;
	int t = 20;
	int count = 0;
	int x = 0;
	int y = 0;
	for(int i = 0; i < m.rows; i++){
		for(int j = 0; j < m.cols; j++){
			cv::Vec3b pixel = m.at<cv::Vec3b>(cv::Point(j,i));
			if (threshold(b,g,r,pixel.val[0],pixel.val[1],pixel.val[2],t)){
				x += j;
				y += i;
				count += 1;
			}
		}
	}
	if (count == 0){
		printf("(0,0)\n");
	}else{
		printf("(%d,%d)\n",x/count,y/count);
	}
	return 0;
}
 	 	 
